pkg/webui/account/views/create-account/index.js

```diff
@@ -91,151 +86,107 @@ const getSuccessMessage = state => {
-
-    if (fetching) {
-      return (
-        <Spinner center>
-          <Message content={sharedMessages.fetching} />
-        </Spinner>
-      )
-    }
-
-    if (Boolean(user)) {
-      return (
-        <Redirect
-          to={{
-            pathname: '/',
-          }}
-        />
-      )
-    }
-
-    return (
-      <React.Fragment>
-        <div className={style.form}>
-          <IntlHelmet title={m.createANewAccount} />
-          <h1 className={style.title}>
-            {siteName}
-            <br />
-            <Message content={m.createANewAccount} component="strong" />
-          </h1>
-          <hr className={style.hRule} />
-          <Form
-            onSubmit={this.handleSubmit}
-            initialValues={initialValues}
-            error={error}
-            validationSchema={validationSchema}
-            horizontal={false}
-          >
-            <Form.Field
-              component={Input}
-              required
-              title={sharedMessages.userId}
-              name="user_id"
-              autoComplete="username"
-              autoFocus
-            />
-            <Form.Field
-              title={sharedMessages.name}
-              name="name"
-              component={Input}
-              autoComplete="name"
-            />
-            <Form.Field
-              required
-              title={sharedMessages.email}
-              component={Input}
-              name="primary_email_address"
-              autoComplete="email"
-            />
-            <Form.Field
-              required
-              title={sharedMessages.password}
-              name="password"
-              type="password"
-              component={Input}
-              autoComplete="new-password"
+const CreateAccount = ({ location }) => {
+  const [error, setError] = useState(undefined)
+  const dispatch = useDispatch()
+
+  const handleSubmit = useCallback(
+    async (values, { setSubmitting }) => {
+      try {
+        setError(undefined)
+        const { user_id, ...rest } = values
+        const { invitation_token = '' } = queryString.parse(location.search)
+        const result = await tts.Users.create({ ids: { user_id }, ...rest }, invitation_token)
+
+        dispatch(
+          push(`/login${location.search}`, {
+            info: getSuccessMessage(result.state),
+          }),
+        )
+      } catch (error) {
+        setSubmitting(false)
+        setError(error)
+      }
+    },
+    [dispatch, location.search],
+  )
+
+  return (
+    <>
+      <div className={style.form}>
+        <IntlHelmet title={m.createANewAccount} />
+        <h1 className={style.title}>
+          {siteName}
+          <br />
+          <Message content={m.createANewAccount} component="strong" />
+        </h1>
+        <hr className={style.hRule} />
+        <Form
+          onSubmit={handleSubmit}
+          initialValues={initialValues}
+          error={error}
+          validationSchema={validationSchema}
```

Comment:
Any reasons for not using https://github.com/TheThingsNetwork/lorawan-stack/blob/feature/1422-account-app-cleanups/pkg/webui/lib/create-password-validation-schema.js for this form?